### PR TITLE
ci(docs-infra): improve deployment process (reduce manual config steps)

### PR DESCRIPTION
**Pull request by gkalpak**
Previously, the stable branch was always deployed to the `v<X>-angular-io-site` Firebase site, which was connected to the `angular.io` domain. Whenever a new major version was released (and became the new stable version), the `angular.io` domain had to be disconnected from the previous Firebase site and be connected to the new `v<Y>-angular-io-site` Firebase site. This was a manual process that involved making changes in the Firebase console and the DNS records.

This commit is part of a new process that reduces the manual steps as follows:
- A new `stable-angular-io-site` Firebase site will be created.
- The `angular.io` domain will be connected to that new Firebase site.
- When deploying from the stable branch, we will deploy to both `stable-angular-io-site` and `v<X>-angular-io-site`.
- When a new major version is released, we will start deploying from the new stable branch to `stable-angular-io-site`, but there will be no need to connect/disconnect the `angular.io` domain.

---
**TODO**

_**Before merging this PR:**_
- [x] Decide whether we want to keep the redirect from `v<X>.angular.io` to `angular.io` (and whether we want to automate it via Firebase config - currently it happens via Firebase console and/or DNS):
  - Getting rid of the redirect simplifies the setup, but changes the current behavior.
  - Keeping the redirect preserves the current behavior, but requires either manual steps (to set/unset the redirect in Firebase console/DNS) or special Firebase config (to redirect all requests to `angular.io` while stable).
  - Also, all options have slightly different behavior/timing when releasing a new major.

  _Discussed in the `docs-infra` meeting on 2021-03-03 and decided to (a) keep the `v<X>.angular.io` to `angular.io` redirect and (b) look into managing the redirects from `v<X>.angular.io` to other sub-domains via the Firebase config._

_**After merging this PR:**_
- [ ] Wait for the changes to be deployed (both on `master` and the patch branch).
- [ ] Connect `angular.io` and `www.angular.io` to the `stable-angular-io-site` Firebase site.
    (Ensure the domains are disconnected from the `v12-angular-io-site` Firebase site.)
- [ ] Remove the redirect from `v13.angular.io` to `rc.angular.io` (in firebase)
- [ ] Remove the redirect from `v12.angular.io` to `angular.io` (in dns)

### CODE REVIEW

File: aio/scripts/deploy-to-firebase.js [hunk 1]
```diff
@@ -118,15 +118,23 @@ function computeDeploymentsInfo(
     stable: {
       deployEnv: 'stable',
       projectId: 'angular-io',
-      siteId: `v${currentBranchMajorVersion}-angular-io-site`,
+      siteId: 'stable-angular-io-site',
```
Comment (petebacondarwin) [inline aio/scripts/deploy-to-firebase.js:121]: Do we need this site to be created (and DNS to be setup??) before merging this PR?

**Review by petebacondarwin — APPROVED** [review]
LGTM!

File: aio/scripts/deploy-to-firebase.js (hunk 1 above)
Comment (gkalpak) [reply aio/scripts/deploy-to-firebase.js:121]: Yes, we do need the site (I have already created :grin:), but we should not change the DNS (i.e. connect the `[www.]angular.io` domain to it) until we have a successful deployment (i.e. after the PR has been merged).

**Review by IgorMinar — APPROVED** [review]
LGTM, thanks for all the tests!

Reviewed-for: docs-infra

**Pull request closed without merging**